Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,349 @@
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using Microsoft.MixedReality.Toolkit.SDK.Input;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Receivers
+{
+    /// <summary>
+    /// An interaction receiver is simply a component that attached to a list of interactable objects and does something
+    /// based on events from those interactable objects.  This is the base abstract class to extend from.
+    /// </summary>
+    public abstract class InteractionReceiver : BaseInputHandler,
+        IMixedRealityFocusChangedHandler,
+        IMixedRealityGestureHandler,
+        IMixedRealityInputHandler,
+        IMixedRealityGestureHandler<Vector2>,
+        IMixedRealityGestureHandler<Vector3>,
+        IMixedRealityGestureHandler<Quaternion>
+
+    {
+        #region Public Members
+        /// <summary>
+        /// List of linked interactable objects to receive events for
+        /// </summary>
+        [SerializeField]
+        [ Tooltip("Target interactable Object to receive events for")]
+        private List<GameObject> interactables = new List<GameObject>();
+
+        public List<GameObject> Interactables
+        {
+            get { return interactables; }
+            set { value = interactables; }
+        }
+
+        /// <summary>
+        /// Is Focus required to receive input events on this GameObject?
+        /// InteractionReceivers handle events for objects indirectly, in order for this class work properly, keep this set to false.
+        /// </summary>
+        public override bool IsFocusRequired
+        {
+            get
+            {
+                return base.IsFocusRequired;
+            }
+
+            protected set
+            {
+                base.IsFocusRequired = value;
+            }
+        }
+
+        /// <summary>
+        /// List of linked targets that the receiver affects
+        /// </summary>
+        [Tooltip("Targets for the receiver to affect")]
+        private List<GameObject> targets = new List<GameObject>();
+
+        public List<GameObject> Targets
+        {
+            get { return targets; }
+            set { value = targets; }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// On start subscribe to all interaction events on elements in the interactables list.
+        /// </summary>
+        protected override void OnEnable()
+        {
+            IsFocusRequired = false;
+            base.OnEnable();
+        }
+
+        /// <summary>
+        /// On disable remove all linked interactables from the delegate functions
+        /// </summary>
+        protected override void OnDisable()
```

Comment:
Do we need this override that just calls the base method?

Reply:
We do not.